From 9c666d0d6039a87f6286014c7d9c7281a5bd9dd1 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Tue, 16 Feb 2021 21:38:14 +0100
Subject: [PATCH] MySQL server settings: Make use of 'loose_' option prefix

Idea by one of the Gentoo MySQL maintainers. This makes MySQL skip over
unsupported options rather than fail, which is important with >=MySQL-8.

See also:
https://dev.mysql.com/doc/refman/8.0/en/option-files.html#option-file-syntax
https://mariadb.com/kb/en/configuring-mariadb-with-option-files/#option-prefixes

BUG: 421922

Thanks-to: Thomas Deutschmann <whissi@gentoo.org>
---
 src/server/storage/mysql-global-mobile.conf | 6 +++---
 src/server/storage/mysql-global.conf        | 6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/server/storage/mysql-global-mobile.conf b/src/server/storage/mysql-global-mobile.conf
index 19649d347..48e150af7 100644
--- a/src/server/storage/mysql-global-mobile.conf
+++ b/src/server/storage/mysql-global-mobile.conf
@@ -65,7 +65,7 @@ innodb_large_prefix=1
 log_error=mysql.err
 
 # print warnings and connection errors (default:1)
-log_warnings=2
+loose_log_warnings=2
 
 # Convert table named to lowercase
 lower_case_table_names=1
@@ -80,10 +80,10 @@ max_connections=256
 # (make sense when having the same query multiple times)
 
 # Memory allocated for caching query results (default:0 (disabled))
-query_cache_size=0
+loose_query_cache_size=0
 
 # Do not cache results (default:1)
-query_cache_type=0
+loose_query_cache_type=0
 
 # Do not use the privileges mechanisms
 skip_grant_tables
diff --git a/src/server/storage/mysql-global.conf b/src/server/storage/mysql-global.conf
index ab4a4a1e3..b67a4ec1e 100644
--- a/src/server/storage/mysql-global.conf
+++ b/src/server/storage/mysql-global.conf
@@ -62,7 +62,7 @@ innodb_log_file_size=64M
 log_error=mysql.err
 
 # print warnings and connection errors (default:1)
-log_warnings=2
+loose_log_warnings=2
 
 # Convert table named to lowercase
 lower_case_table_names=1
@@ -77,10 +77,10 @@ max_connections=256
 # (make sense when having the same query multiple times)
 
 # Memory allocated for caching query results (default:0 (disabled))
-query_cache_size=0
+loose_query_cache_size=0
 
 # Do not cache results (default:1)
-query_cache_type=0
+loose_query_cache_type=0
 
 # Do not use the privileges mechanisms
 skip_grant_tables
-- 
2.30.1

